elser/README.md

```diff
@@ -0,0 +1,133 @@
+to _enable_ Block Max WAND and `true` to _disable_).
+
+The `text_expansion` query rewrites to a compound weighted term query using the token weights
+retuned by evaluating the [ELSER](https://www.elastic.co/guide/en/machine-learning/master/ml-nlp-elser.html) model. For benchmarking, evaluating the NLP model is not necessary nor is it desired as the 
+purpose is to test the generated query. The query terms are pre-generated by the ELSER model
+and hard coded into the searches. 
+
+
+The BM25 match queries are present to establish a benchmark for comparrison with the
+weighted term queries. 
+
+
+### Generating the queries
+The weighted tokens are found by evaluating ELSER with the [Inference API](https://www.elastic.co/guide/en/elasticsearch/reference/current/infer-trained-model.html) passing the query text.
+The response of the inference call contains the weighted tokens which are converted to
+the query with the following `jq` command:
+
+```
+
+ jq '.inference_results | .[0] | .predicted_value | { "query": {"bool": { "should":  [ to_entries | .[] | {term: { "ml.tokens": { value: .key, boost: .value } } } ] } } } ' infer_output.json
+
+```
+In this example the file `infer_output.json` contains the response of the inference call.
+
+
+The benchmark considers queries with different numbers of terms. Again these queries are created from the inference 
+response, the following script creates the operations in `operations/weighted-token-queries.json`. The first argument to the script 
+is a file containing the inference response. 
+
+```
+#!/bin/sh
+
+if [ -z $1 ]
+then 
+	echo "Missing input file: usgae ./gen_queries.sh input_json"; 
+	exit 1
+fi
+
+INPUT_FILE=$1
+
+for NUM_TOKENS in 10 20 30 40 50
+do
+
+	TRACK_TOTAL_HITS=true
+	QUERY_NAME=$NUM_TOKENS-weighted-terms-maxwand-disabled
+
+	jq --arg track_total $TRACK_TOTAL_HITS --argjson num_tokens $NUM_TOKENS --arg name $QUERY_NAME '.inference_results | .[0] | .predicted_value | { "name": $name,  "operation-type": "search", "body": { "track_total_hits": $track_total, "query": {"bool": { "should":  [ to_entries | .[0:$num_tokens] | .[] | {term: { "ml.tokens": { value: .key, boost: .value } } } ] } } } }' $INPUT_FILE
+
+	echo ','
+
+	TRACK_TOTAL_HITS=false
+	QUERY_NAME=$NUM_TOKENS-weighted-terms-maxwand-enabled
+
+	jq --arg track_total $TRACK_TOTAL_HITS --argjson num_tokens $NUM_TOKENS --arg name $QUERY_NAME '.inference_results | .[0] | .predicted_value | { "name": $name,  "operation-type": "search", "body": { "track_total_hits": $track_total, "query": {"bool": { "should":  [ to_entries | .[0:$num_tokens] | .[] | {term: { "ml.tokens": { value: .key, boost: .value } } } ] } } } }' $INPUT_FILE
+
+	echo ','
+done 
+
+```
+
+### Preparing the Dataset
+<!-- TODO Ingest MS Marco details. -->
+
+
+Once restored the MS Marco passage documents need to be run through ELSER to generate the tokens.
+This is done by reindexing through an ingest pipeline containing the inference processor.
```

Comment:
Not sure I understand this part, are you saying that the index is generated from a snapshot? Can we index the raw documents instead and add a version where the weighted terms are merged with the original documents?

Reply:
I am trying describing the process of creating the test corpus with ELSER tokens. I've refined the README, hope it is clearer now.